src/editor/editor.js

```diff
@@ -221,33 +222,16 @@ export default class Editor {
 	 * @returns {Promise} A promise which resolves once the initialization is completed.
 	 */
 	initPlugins() {
-		const that = this;
 		const config = this.config;
 
-		return loadPlugins()
-			.then( loadedPlugins => {
-				return initPlugins( loadedPlugins, 'init' )
-					.then( () => initPlugins( loadedPlugins, 'afterInit' ) );
-			} )
-			.then( () => this.fire( 'pluginsReady' ) );
-
-		function loadPlugins() {
-			const plugins = config.get( 'plugins' ) || [];
-			const removePlugins = config.get( 'removePlugins' ) || [];
-			const extraPlugins = config.get( 'extraPlugins' ) || [];
-
-			return that.plugins.load( plugins.concat( extraPlugins ), removePlugins );
-		}
-
-		function initPlugins( loadedPlugins, method ) {
-			return loadedPlugins.reduce( ( promise, plugin ) => {
-				if ( !plugin[ method ] ) {
-					return promise;
-				}
+		return Promise.resolve()
+			.then( () => {
+				const plugins = config.get( 'plugins' ) || [];
+				const removePlugins = config.get( 'removePlugins' ) || [];
+				const extraPlugins = config.get( 'extraPlugins' ) || [];
 
-				return promise.then( plugin[ method ].bind( plugin ) );
-			}, Promise.resolve() );
-		}
+				return this.plugins.init( plugins.concat( extraPlugins ), removePlugins );
+			} );
```

Comment:
I believe it could be simplified now to:
```js
	initPlugins() {
		const plugins = this.config.get( 'plugins' ) || [];
		const removePlugins = this.config.get( 'removePlugins' ) || [];
		const extraPlugins = this.config.get( 'extraPlugins' ) || [];

		return this.plugins.init( plugins.concat( extraPlugins ), removePlugins );
	}
```